src/test/java/uk/gov/hmcts/reform/pcqloader/services/BlobStorageManagerTest.java

```diff
@@ -170,8 +170,23 @@ void testCollectBlobFileNamesEmptyListSuccess() {
     }
 
     @Test
-    void testCollectBlobFileNamesMissingName() {
-        BlobItem blobItem1 = new BlobItem().setDeleted(false).setIsPrefix(null);
+    void testCollectBlobFileNamesWithNameNull() {
+        BlobItem blobItem1 = new BlobItem().setDeleted(false).setIsPrefix(false).setName(null);
+        var blobs = Arrays.asList(blobItem1);
+
+        when(pcqContainer.listBlobsByHierarchy(ROOT_FOLDER)).thenReturn(pageIterableBlobs);
+        when(pageIterableBlobs.iterator()).thenReturn(blobs.iterator());
+
+        testBlobStorageManager = new BlobStorageManager(blobStorageProperties, blobServiceClient, zipFileUtils);
+        List<String> response = testBlobStorageManager.collectBlobFileNamesFromContainer(pcqContainer);
+
+        verify(pageIterableBlobs, times(1)).iterator();
+        Assertions.assertEquals(0, response.size(), "No files added as no name was provided");
```

Comment:
If I am not mistaken, message should tell what went wrong. In this case it is stating the right fact, meaning that if this test fails, you'll get message - no files added as no name was provided.. which basically tells that it was fine. Better message would be - "files were added, but we didn't expect any" or smth. like that.